Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java

```diff
@@ -143,11 +143,13 @@ protected void downloadSubmittedCodes() throws IOException {
     @Override
     protected void downloadProvidedCodes() throws IOException {
         LanguageEnv languageEnv = getLanguageEnv();
-        try (FileResource zip = problemServiceDriver.downloadProvidedCodes(
-                getProblem().getId(), languageEnv.getName(), languageEnv.getProvidedCodesFileId())) {
-            ZipUtils.unzipToDestination(zip.getInputStream(), getSourceRootPath());
-        } catch (NotFoundException ignored) {
-            logger.info("No providedCodes.");
+        if (languageEnv.getProvidedCodesFileId() != null) {
+            try (FileResource zip = problemServiceDriver.downloadProvidedCodes(
+                    getProblem().getId(), languageEnv.getName(), languageEnv.getProvidedCodesFileId())) {
```

Comment:
Please declare a variable 'providedCodesFileId' for ```languageEnv.getProvidedCodesFileId();```

Reply:
Thank you but I don't think this helps.

Reply:
Ok. You can merge this pr.